lib/traits/device_widget_builder.dart

```diff
@@ -0,0 +1,97 @@
+import 'package:flutter/material.dart';
+import 'package:yonomi_device_widgets/providers/battery_level_provider.dart';
+import 'package:yonomi_device_widgets/providers/lock_provider.dart';
+import 'package:yonomi_device_widgets/providers/power_trait_provider.dart';
+import 'package:yonomi_device_widgets/providers/thermostat_provider.dart';
+import 'package:yonomi_device_widgets/traits/device_widget_factory.dart';
+import 'package:yonomi_device_widgets/ui/widget_style_constants.dart';
+import 'package:yonomi_platform_sdk/yonomi-sdk.dart';
+
+class DeviceWidgetBuilder {
+  final _primaryTraitWidgets = <Widget>[];
+  final _traitWidgets = <Widget>[];
+
+  DeviceWidgetBuilder withPrimaryTrait(Trait trait,
+      {padding = const EdgeInsets.all(8.0),
+      margins = const EdgeInsets.only(bottom: 8.0)}) {
+    _primaryTraitWidgets
+        .add(_card(_createTraitWidget(trait), padding, margins));
+
+    return this;
+  }
+
+  DeviceWidgetBuilder withTrait(trait,
```

Comment:
This doesn't need to return anything, maybe rename to addTrait or something like that.

Reply:
It does if it's going to be used to chain the methods together.